From c150bc4cc27d560970fd34cf2347142325e7704a Mon Sep 17 00:00:00 2001
From: Michal Sekletar <msekleta@redhat.com>
Date: Tue, 21 Sep 2021 15:01:19 +0200
Subject: [PATCH] udev/net-setup-link: change the default MACAddressPolicy to
 "none"

While stable MAC address for interface types that don't have the
address provided by HW could be useful it also breaks LACP based bonds.
Let's err on the side of caution and don't change the MAC address from
udev.

RHEL-only

Related: #2138081
---
 man/systemd.link.xml                       | 2 +-
 network/99-default.link                    | 2 +-
 test/fuzz/fuzz-link-parser/99-default.link | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/man/systemd.link.xml b/man/systemd.link.xml
index cc55b02b18..69e0ac779c 100644
--- a/man/systemd.link.xml
+++ b/man/systemd.link.xml
@@ -1117,7 +1117,7 @@
 
       <programlisting>[Link]
 NamePolicy=kernel database onboard slot path
-MACAddressPolicy=persistent</programlisting>
+MACAddressPolicy=none</programlisting>
     </example>
 
     <example>
diff --git a/network/99-default.link b/network/99-default.link
index 429ac31e80..083bdb5659 100644
--- a/network/99-default.link
+++ b/network/99-default.link
@@ -17,4 +17,4 @@ OriginalName=*
 [Link]
 NamePolicy=keep kernel database onboard slot path
 AlternativeNamesPolicy=database onboard slot path
-MACAddressPolicy=persistent
+MACAddressPolicy=none
diff --git a/test/fuzz/fuzz-link-parser/99-default.link b/test/fuzz/fuzz-link-parser/99-default.link
index feb5b1fbb0..3d755898b4 100644
--- a/test/fuzz/fuzz-link-parser/99-default.link
+++ b/test/fuzz/fuzz-link-parser/99-default.link
@@ -9,4 +9,4 @@
 
 [Link]
 NamePolicy=keep kernel database onboard slot path
-MACAddressPolicy=persistent
+MACAddressPolicy=none
